### PR TITLE
feat(nepserver): add mqtt support for home-assistant

### PR DESCRIPTION
**Pull request by wrenix**
add also following function:
  - prometheus HELP (and TYPE) 
  - localhost:8080/data.json to get all values for debugging (with the last date when values was recieved)
  - extract parsing and formating to extra python class

---
here rendered new:
- [README](https://github.com/wrenix/NEP-Inverter-Reversing/blob/main/README.md)
- [docs/setup_server.md](https://github.com/wrenix/NEP-Inverter-Reversing/blob/main/docs/setup_server.md)

### CODE REVIEW

**Comment by Sn0w3y** [comment]
Just tell me, if the PR os ready :) Will merge it after it :)

**Comment by Sn0w3y** [comment]
@wrenix please Review the How-to.md also i added you as Collaborator to make things easier for you aswell :)

I please you to keep up with GitHubs best practices as you already do and always add some Descriptions of how to use our Projects :)

**Comment by wrenix** [comment]
for me, it is ready ;)

**Comment by Sn0w3y** [comment]
> for me, it is ready ;)

Could you please review the How-To aswell if it is clear and if it works as expected? :)

**Comment by Sn0w3y** [comment]
> for me, it is ready ;)
> 
> the dnsmasq instruduction locks good for me (i use another dns tool ...)

could you re-write the How to or check if it works as expected ? :)

Either we refactor our code later on to include a functionality to include a DNS Automatically - will work on this if my Inverter is Back online again :P

File: how-to's/setup_server.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+3. **Install DNS Server Software:** Use `dnsmasq` for its simplicity and flexibility. Install it by running:
+   ` sudo apt update && sudo apt install dnsmasq `
+
+4. **Configure `dnsmasq`:** Create a fake DNS entry for `www.nepviewer.net` by editing the `dnsmasq` configuration.
+   Backup the original configuration file:
+   `sudo mv /etc/dnsmasq.conf /etc/dnsmasq.conf.backup`
+   Create a new configuration file with:
+   `sudo nano /etc/dnsmasq.conf`
+   Add the following lines, replacing `raspberry_pi_ip` with your Raspberry Pi’s IP address and `server_ip` with your server's IP address:
+   `listen-address=127.0.0.1,raspberry_pi_ip
```
Comment (wrenix) [inline how-to's/setup_server.md:23]:
````suggestion
   ```
   listen-address=127.0.0.1,raspberry_pi_ip
   address=/www.nepviewer.net/server_ip
   ```
````
```

**Comment by wrenix** [comment]
i believe your DNS-Doku is good -> my current solution disable on my wifi all other DNS .... (which is okay for me, i like to block everything on my iot wifi)

**Comment by wrenix** [comment]
Add some more Docu ... so i am ready, now ;)

**Pull request merged**